www/js/index.js

```diff
@@ -1,49 +1,225 @@
+            beforeStart: setLoadingState
+        });
+
+        $.when(stopsPromise, departuresPromise)
+            .done(function(stopsResult, departuresResult) {
+                // stopResults object format example:
+                // `[{"rows":[{"id":15,"calendar":"WEEKDAY","time":"05:50"}],"rowsAffected":0}, 'statusText', jqXHR]`
+                var stops = stopsResult[0].rows;
+                // departuresResult object format example:
+                // `[{"rows":[{"id":1,"name":"TICEN","sequence":1,"route_id":35}],"rowsAffected":0}, 'statusText', jqXHR]`
+                var departures = departuresResult[0].rows;
+
+                renderStops(stops);
+                renderDepartures(departures);
+            })
+            .always(removeLoadingState);
+    };
+
+    function backToListView() {
+        $('.list-view').removeClass('hide');
+        $('.detail-view').addClass('hide');
+        renderDepartures([]);
+        renderStops([]);
+    };
+
+    /**
+     * Example of result:
+     * @param routes JSON object following structure:
+     *   `{"id":22,"shortName":"131","longName":"AGRONÔMICA VIA GAMA D'EÇA","lastModifiedDate":"2009-10-26T02:00:00+0000","agencyId":9}`
+     */
+    function renderRoutes(routes) {
+        var template = $('#routes-result-tmpl').html();
+        Mustache.parse(template);
+        var rendered = Mustache.render(template, {
+            routes: routes,
+            empty: routes.length == 0
+        });
+
+        $('.routes-panel')
+            .empty()
+            .append(rendered);
+    };
+
+    function searchRoutesByStopName(name, args) {
+
+        args = $.extend({}, {
+            beforeStart: emptyFunc
+        }, args);
+        var beforeStart = args.beforeStart;
+        var options = $.extend({}, DEFAULT_AJAX_OPTIONS, {
+            url: ROUTES_BY_STOP_NAME_ENDPOINT,
+            data: JSON.stringify({
+                params: {
+                    stopName: name
+                }
+            }),
+            success: function(data, textStatus, jqXHR) {
+                // Example of returned data format:
+                // {"rows": [{"id":22,"shortName":"131","longName":"AGRONÔMICA VIA GAMA D'EÇA","lastModifiedDate":"2009-10-26T02:00:00+0000","agencyId":9}], "rowsAffected":0}
+                var rows = data.rows;
+                renderRoutes(rows);
+            }
+        });
+
+        beforeStart();
+        return $.ajax(options)
+    };
+
+    function init() {
+        var $routeSearchForm = $('form.route-search-form');
+
+        $routeSearchForm.submit(function(e) {
+
+            e.preventDefault();
+            var $location = $('[name="location"]', $routeSearchForm),
+                location = $location.val().trim();
+
+            if (!location) {
+                $routeSearchForm.prepend($('#no-input-error-tmpl').html());
+            } else {
+                var promise = searchRoutesByStopName('%' + location + '%', {
```

Comment:
No strong reason actually... I just thought it would be easier to figure that `searchRoutesByStopName` returns a `Promise` ... 

It indeed could be useing chained calls like:

```javascript
searchRoutesByStopName('%' + location + '%', {
        beforeStart: function() {
            $routeSearchForm.addClass('loading');
        }
    })
    .always(function() {
        $routeSearchForm.removeClass('loading');
    });
```